ui/admin/app/components/form/storage-bucket/aws/index.hbs

```diff
@@ -453,30 +453,41 @@
   {{/if}}
 
   {{! worker_filter }}
-  <Hds::Form::TextInput::Field
-    name='worker_filter'
-    @isRequired={{true}}
-    @value={{@model.worker_filter}}
-    @isInvalid={{@model.errors.worker_filter}}
-    disabled={{form.disabled}}
-    {{on 'input' (set-from-event @model 'worker_filter')}}
-    as |F|
-  >
-    <F.Label>{{t 'resources.storage-bucket.form.worker_filter.label'}}</F.Label>
-    <F.HelperText>
-      {{t 'resources.storage-bucket.form.worker_filter.help'}}
-      <Hds::Link::Inline @href={{doc-url 'storage-bucket.worker-filter'}}>
-        {{t 'actions.learn-more'}}
-      </Hds::Link::Inline>
-    </F.HelperText>
-    {{#if @model.errors.worker_filter}}
-      <F.Error data-test-worker-filter-error as |E|>
-        {{#each @model.errors.worker_filter as |error|}}
-          <E.Message>{{error.message}}</E.Message>
-        {{/each}}
-      </F.Error>
-    {{/if}}
-  </Hds::Form::TextInput::Field>
+  {{#let (unique-id) (unique-id) as |labelId helpId|}}
+    <Hds::Form::Fieldset
+      aria-labelledby={{labelId}}
+      aria-describedby={{helpId}}
+      class='storage-bucket-worker-filter'
+      @isRequired={{true}}
+      as |F|
+    >
+      <F.Legend id={{labelId}}>{{t
+          'resources.storage-bucket.form.worker_filter.label'
+        }}</F.Legend>
+      <F.HelperText id={{helpId}}>
+        {{t 'resources.storage-bucket.form.worker_filter.help'}}
+        <Hds::Link::Inline @href={{doc-url 'storage-bucket.worker-filter'}}>
+          {{t 'actions.learn-more'}}
+        </Hds::Link::Inline>
+      </F.HelperText>
+      <F.Control>
+        {{#if form.disabled}}
+          <Hds::CodeBlock
+            @language='bash'
+            @value={{@model.worker_filter}}
+            @hasCopyButton={{true}}
+            data-test-worker-filter
+          />
+        {{else}}
+          <WorkerFilterGenerator
```

Comment:
I am not sure I understand the question. Could you elaborate more please?

Reply:
There is no limit on the number of expressions that a user can add to the generator so a page could, in theory, look like the screenshot below and I wonder if we want to allow that:
<img width="1030" alt="Screenshot 2024-12-09 at 6 42 05 PM" src="https://github.com/user-attachments/assets/06d49519-1d6a-4c84-a6ad-92123d2ecccc">

Reply:
I don't think we should limit the number of expressions a user can add. Also, not sure yet how users feel with UX, but in case we want to limit something, we can limit the high the code block uses, therefore, if the code content is big, it will be scrollable.
[HDS docs about limit height](https://helios.hashicorp.design/components/code-block?tab=code#limit-height)

Reply:
Oh I see. I would agree with @calcaide that we shouldn't limit the user's input and I like the idea of a max height for the code block component. I will a max height after reaching out to design about this and the other spacing question.

Reply:
Added a max height for the code blocks.

Reply:
That idea works too! I just wanted to avoid what the UI looks such as the case provided above.